src/features/layers_in_area/atoms/areaLayersLegendsAndSources.ts

```diff
@@ -84,10 +84,13 @@ export const areaLayersLegendsAndSources = createAtom(
         const newState = new Map(state);
         requestedLayers.forEach((layerId) => {
           const layerDetails = layersDetailsData.get(layerId);
-          const layerSource = layerDetails ? convertDetailsToSource(layerDetails) : null;
+          const layerSourceFromResource = layerDetails
+            ? convertDetailsToSource(layerDetails)
+            : null;
+          const existedLayerSource = state.get(layerId)?.data ?? null;
           newState.set(layerId, {
             error: layersDetailsError,
-            data: layerSource,
+            data: layerSourceFromResource || existedLayerSource,
             isLoading: false,
           });
         });
```

Comment:
I could not understand the logic of this code.
I understand that we change the behavior here when the backend did not send us the details for requested layer.
So we ask details for layers A, B, C - back-end send data details only for A, C.
Before that in such case we set layer data details to null (sound logical for me) - now - we trying to keep last data that we get from backend, but why?

Reply:
@Akiyamka  When we switch between cached layers (previously enabled) - they don't get here
So we have source for layer B in `layersSourcesAtom` but we don't have source for it in resource response

Reply:
Changed elsewhere